src/esssans/normalization.py

```diff
@@ -110,63 +112,96 @@ def transmission_fraction(
 
     Parameters
     ----------
-    data_transmission_monitor:
-        The transmission monitor counts from the measurement run (depends on
-        wavelength).
-    direct_incident_monitor:
-        The incident monitor counts from the direct run (depends on wavelength).
-    direct_transmission_monitor:
-        The transmission monitor counts from the direct run (depends on wavelength).
+    incident_monitor:
+        The incident monitor data (depends on wavelength).
+    transmission_fraction:
+        The transmission fraction (depends on wavelength).
     direct_beam:
-        The DataArray containing the direct beam function (depends on wavelength).
+        The direct beam function (depends on wavelength).
     uncertainties:
         The mode for broadcasting uncertainties. See
         :py:class:`UncertaintyBroadcastMode` for details.
 
     Returns
     -------
     :
-        Wavelength-dependent term for the denominator of the SANS I(Q) normalization.
+        Wavelength-dependent term
+        (incident_monitor * transmission_fraction * direct_beam) to be used for
+        the denominator of the SANS I(Q) normalization.
         Used by :py:func:`iofq_denominator`.
-    """  # noqa: E501
-    denominator = (
-        data_transmission_monitor
-        * direct_incident_monitor
-        / direct_transmission_monitor
-    )
-
-    # We need to remove the variances because the broadcasting operation between
-    # solid_angle (pixel-dependent) and monitors (wavelength-dependent) will fail.
-    # The direct beam may also be pixel-dependent. In this case we need to drop
-    # variances of the other term already before multiplying by solid_angle.
+    """
+    out = incident_monitor * transmission_fraction
     if direct_beam is not None:
         broadcast = _broadcasters[uncertainties]
-        denominator = direct_beam * broadcast(denominator, sizes=direct_beam.sizes)
+        # TODO: Do we need an additional check for the case where the direct beam
+        # could be bin centers and the transmission fraction is bin edges? In that case
+        # we would want to raise instead of broadcasting.
+        out = direct_beam * broadcast(out, sizes=direct_beam.sizes)
     # Convert wavelength coordinate to midpoints for future histogramming
-    # if wavelength_to_midpoints:
-    denominator.coords['wavelength'] = sc.midpoints(denominator.coords['wavelength'])
-    return NormWavelengthTerm[RunType](denominator)
+    out.coords['wavelength'] = sc.midpoints(out.coords['wavelength'])
+    return out
+
+
+def iofq_norm_wavelength_term_sample(
+    incident_monitor: CleanMonitor[SampleRun, Incident],
+    transmission_fraction: TransmissionFraction[SampleTransmissionRun],
+    direct_beam: Optional[CleanDirectBeam],
+    uncertainties: UncertaintyBroadcastMode,
+) -> NormWavelengthTerm[SampleRun]:
+    """
+    Compute the wavelength-dependent contribution to the denominator term for the I(Q)
+    normalization, for the sample run.
```

Comment:
Not sure what to do about the docstring here. Do we just duplicate between `iofq_norm_wavelength_term_sample` and `iofq_norm_wavelength_term_background`?

Reply:
So your problem is that `TransmissionFraction` depends on `[Sample|Background]TransmissionRun`, and you have this duplication (and the helper function above) do get away from that?

I feel this should be handled somehow when *computing* `TransmissionFraction`, not when using it.

Would a nested generic help?

```python
def transmission_fraction(
    sample_incident_monitor: CleanMonitor[Transmission[RunType], Incident],
    sample_transmission_monitor: CleanMonitor[Transmission[RunType], Transmission],
    direct_incident_monitor: CleanMonitor[DirectRun, Incident],
    direct_transmission_monitor: CleanMonitor[DirectRun, Transmission],
) -> TransmissionFraction[RunType]:

where `RunType` would not be `SampleRun` or `BackgroundRun`?

Reply:
To put it differently:
- The IofQ computation should not have to know how the transmission fraction was computed. The transmission fraction is the transmission fraction, and should not carry the difference between `SampleRun` and `SampleTransmissionRun` forward into the rest of the workflow.
- Splitting providers that do the same thing will lead to misleading graphs: Readers will think the operations do different things, so we should avoid this as much as possible.